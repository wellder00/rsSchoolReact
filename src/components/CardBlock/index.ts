export { default as CardBlock } from './CardBlock';
