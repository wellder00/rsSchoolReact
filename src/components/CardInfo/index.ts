export { default as CardInfo } from './CardInfo';
