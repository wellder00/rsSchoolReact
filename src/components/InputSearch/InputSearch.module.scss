.search {
  transition:
    border-color 0.3s,
    box-shadow 0.3s;
  outline: none;
  box-shadow: 0 0 20px rgba(0, 0, 0, 0);
  border: 2px solid #3498db;
  border-radius: 5px;
  background-color: #e8edf1;
  padding: 10px;
  font-size: 16px;
}

.search:focus {
  box-shadow: 0 0 20px rgba(0, 0, 0, 0.4);
  border-color: #01f801;
}
