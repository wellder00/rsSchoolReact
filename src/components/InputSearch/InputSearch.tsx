import React from 'react';
import styles from './InputSearch.module.scss';

type Props = {
  placeholder: string;
  className: string;
  pokemonName: string;
  onChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  getInputValue: () => void;
};

const InputSearch: React.FC<Props> = ({
  placeholder,
  className,
  onChange,
  getInputValue,
  pokemonName,
}) => {
  const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
    if (event.key === 'Enter') {
      getInputValue();
    }
  };

  return (
    <input
      className={styles[className]}
      type="text"
      placeholder={placeholder}
      value={pokemonName}
      onChange={onChange}
      onKeyDown={handleKeyDown}
    />
  );
};

export default InputSearch;
