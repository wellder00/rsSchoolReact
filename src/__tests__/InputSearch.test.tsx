import { render, screen } from '@testing-library/react';
import { InputSearch } from '../components/InputSearch';
import { ChangeEvent } from 'react';
import '@testing-library/jest-dom';

describe('InputSearch', () => {
  test('renders InputSearch component', () => {
    const component = render(
      <InputSearch
        placeholder={''}
        className={''}
        pokemonName={''}
        onChange={function (event: ChangeEvent<HTMLInputElement>): void {
          throw new Error('Function not implemented.');
        }}
        getInputValue={function (): void {
          throw new Error('Function not implemented.');
        }}
      />
    );
    expect(component).toMatchSnapshot();
    expect(screen.getByPlaceholderText('')).toBeInTheDocument();
    expect(screen.getByTestId('input-test-id')).toBeInTheDocument();
  });
});
