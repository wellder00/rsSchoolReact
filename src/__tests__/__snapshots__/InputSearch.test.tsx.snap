// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Tests for the Card component > Testing home page 1`] = `
<DocumentFragment>
  <input
    class="_someClass_823600"
    placeholder="text"
    type="text"
    value=""
  />
</DocumentFragment>
`;
